### ISSUE TITLE
Enable more non-default warnings

### ISSUE THREAD

**Issue by TheJJ** [SFTtech/openage#919]
Although we've long hat `-Wall` and `-Wextra`, there is more useful warnings:

https://kristerw.blogspot.de/2017/09/useful-gcc-warning-options-not-enabled.html

Which of those should we activate?

**Comment by castilma** [comment]
dumb question: why not all?
partial answer: some options are only available in gcc 7. can we tell cmake to not pass unknown options to older versions?

**Comment by TheJJ** [comment]
Indeed. And while enabling them, we also have to fix problems discovered by them so the output remains warning-free. Not all options are that useful for us, for example `-Wold-style-cast` may require a lot of work. `-Wshadow` may also be undesirable in some rare cases, but of course the shadowing should be fixed instead.

In general it would be good to activate as many of the warnings as possible, but it requires some work.
